chain/epoch-manager/src/lib.rs

```diff
@@ -1357,15 +1350,21 @@ impl EpochManager {
     }
 
     pub fn get_shard_layout(&self, epoch_id: &EpochId) -> Result<ShardLayout, EpochError> {
-        let protocol_version = self.get_epoch_info(epoch_id)?.protocol_version();
-        Ok(self.get_shard_layout_from_protocol_version(protocol_version))
+        let epoch_info = self.get_epoch_info(epoch_id)?;
+        if let Some(shard_layout) = epoch_info.shard_layout() {
```

Comment:
Can you add a comment please? 
Just to make sure I understand correctly, this is equivalent to the protocol version check, is that right?